### PR TITLE
Fix/DELETE CollectionLinkedRegistrationsRelationship 500 Error [EMB-571]

### PR DESCRIPTION
**Pull request by pattisdr**
## Purpose

You get a 500 with a DELETE request to the CollectionLinkedRegistrationsRelationship -
 
## Changes

Have CollectionLinkedRegistrationsRelationship inherit from CollectionLinkedNodesRelationship instead of LinkedRegistrationsRelationship and the CollectionMixin.  That view contains everything it needs, besides the overrides in the view itself (allows it to call Collection.remove_object` instead of Collection.perform_destroy which isn't a thing. 

## QA
None needed - but a DELETE request http://localhost:8000/v2/collections/9jdgr/relationships/linked_registrations/ with request body:
```
{
	"data": [
		{"id": "s6teh", "type": "linked_registrations"}
	]
}
``` 
should succeed.

## Ticket

https://openscience.atlassian.net/browse/EMB-571

### CODE REVIEW

File: api/collections/views.py [hunk 1]
```diff
@@ -913,7 +912,6 @@ class CollectionLinkedRegistrationsRelationship(LinkedRegistrationsRelationship,
     )
 
     serializer_class = CollectedRegistrationsRelationshipSerializer
-    view_category = 'collections'
```
Comment (pattisdr) [inline api/collections/views.py]: Already exists on "CollectionLinkedNodesRelationship"

File: api/collections/views.py [hunk 2]
```diff
@@ -842,7 +841,7 @@ def get_object(self):
         return obj
 
 
-class CollectionLinkedRegistrationsRelationship(LinkedRegistrationsRelationship, CollectionMixin):
+class CollectionLinkedRegistrationsRelationship(CollectionLinkedNodesRelationship):
```
Comment (pattisdr) [inline api/collections/views.py]: We could more to DRY up some of these related views - `LinkedNodesRelationship` views and `LinkedRegistrationsRelationship` and similar. CollectionLinkedRegistrationsRelationship and CollectionLinkedNodesRelationship and CollectionLinkedPreprintsRelationship...

But this seems out of scope of this bugfix and would require more time to make sure we have proper test coverage before we do a wide sweep of refactoring.

File: api_tests/collections/test_views.py [hunk 3]
```diff
@@ -3204,6 +3204,29 @@ def test_delete_not_present(
         assert len(res.json['data']) + \
             len(reg_res.json['data']) == number_of_links
 
+    def test_delete_linked_registration(
+            self, app, make_payload, url_private_linked_regs, user_two,
+            collection_private, user_one, registration_private):
+
+        payload = {
+            'data': [
+                {'id': registration_private._id, 'type': 'linked_registrations'}
```
Comment (aaxelb) [inline api_tests/collections/test_views.py]: Given #8917, should we test with `'type': 'registrations'` (version >= 2.13) as well?

Comment (pattisdr) [reply api_tests/collections/test_views.py]: ah, thanks, I missed that this had gone in.

It doesn't look like `type` is enforced though for DELETE requests - we're normally enforcing type checking on the serializer, but deleting is handled on the view [CollectionLinkedNodesRelationship.perform_destroy](https://github.com/CenterForOpenScience/osf.io/blob/develop/api/collections/views.py#L816).  We just loop through the id's in data and remove them.  I could enforce type on this view, I just don't know if this is the proper place to put it/do we need to be strict about DELETE's?

Comment (aaxelb) [reply api_tests/collections/test_views.py]: Strictly speaking I'd expect a 400 if I tried to delete a `type: 'donut'`, and that sort of logic should be implemented only once (not sure where)... but I think adding that is pretty low priority. I'm content with just the test 👍

**Pull request merged**